libbeat/processors/move_fields/move_fields.go

```diff
@@ -0,0 +1,120 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package move_fields
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	"github.com/elastic/beats/v7/libbeat/processors/checks"
+	jsprocessor "github.com/elastic/beats/v7/libbeat/processors/script/javascript/module/processor"
+	"github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+func init() {
+	processors.RegisterPlugin("move_fields",
+		checks.ConfigChecked(NewMoveFields, checks.AllowedFields(
+			"parent_path", "from", "to", "exclude", "ignore_from_not_found",
+		), checks.RequireFields("to")))
+	jsprocessor.RegisterPlugin("MoveFields", NewMoveFields)
+}
+
+type moveFieldsConfig struct {
+	ParentPath         string   `config:"parent_path"`
+	From               []string `config:"from"`
+	IgnoreFromNotFound bool     `config:"ignore_from_not_found"`
+	To                 string   `config:"to"`
+	Exclude            []string `config:"exclude"`
+
+	excludeMap map[string]struct{}
+}
+
+type moveFields struct {
+	config moveFieldsConfig
+}
```

Comment:
Aside the changes I mentioned on the documentation, I noticed another improvement we can make:
 - `excludeMap` is used at runtime by `moveFields`, it's not really a configuration field, so we can move it to `moveFields`.
 - I also ordered the fields by type and alphabetically.

```suggestion
type moveFieldsConfig struct {
	Exclude        []string `config:"exclude"`
	Fields         []string `config:"fields"`
	From           string   `config:"from"`
	To             string   `config:"to"`
	IgnoreMissing bool     `config:"ignore_missing"`
}

type moveFields struct {
	config     moveFieldsConfig
	excludeMap map[string]struct{}
}
```